### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by stephanie1997-kek**
Create sign up page

### CODE REVIEW

**Comment by stephanie1997-kek** [comment]
ready

File: app/assets/stylesheets/users.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (manhnt-0960) [inline app/assets/stylesheets/users.scss]: please delete this empty file

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,19 @@
 class ApplicationController < ActionController::Base
+  before_action :switch_locale
+  # it run around all action run. this is filter action technical
+
+  def default_url_options
```
Comment (manhnt-0960) [inline app/controllers/application_controller.rb:6]: move this method down private

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for user
+  	gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
+  	gravatar_url = "#{Settings.links.gravatar}#{gravatar_id}"
+  	image_tag gravatar_url, alt: user.name, class: "gravatar"
```
Comment (manhnt-0960) [inline app/helpers/users_helper.rb]:
```suggestion
    gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
    gravatar_url = "#{Settings.links.gravatar}#{gravatar_id}"
    image_tag gravatar_url, alt: user.name, class: "gravatar"
```

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      <%= f.label :email, "Email" %>
+      <%= f.email_field :email, class: "form-control" %>
+      <%= f.label :password, "Password" %>
+      <%= f.password_field :password, class: "form-control" %>
+      <%= f.label :password_confirmation, t(".confirmation") %>
+      <%= f.password_field :password_confirmation, class: "form-control" %>
+      <%= f.submit t(".signup_create"), class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
```
Comment (manhnt-0960) [inline app/views/users/new.html.erb]: add endline

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, @user.name) %>
+<div class="row">
+  <aside class="col-md-4">
+  	<section class="user_info">
+  	  <h1>
+  	  	<%= gravatar_for @user %>
+  	  	<%= @user.name %>
+  	  </h1>
+  	</section>
```
Comment (manhnt-0960) [inline app/views/users/show.html.erb]: please check indent again

**Comment by stephanie1997-kek** [comment]
ready

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<% provide(:title, t(".signup_title")) %>
```
Comment (saobv-2223) [inline app/views/users/new.html.erb]: remove ()

**Comment by stephanie1997-kek** [comment]
ready

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::Base
+  private
+
+  before_action :switch_locale
```
Comment (vanvtt-0952) [inline app/controllers/application_controller.rb:2]: move this method up private code

**Comment by stephanie1997-kek** [comment]
ready

**Pull request merged**